### ISSUE TITLE
hasOwnProperty not defined globally, but is available

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/TypeScript#13162]
**TypeScript Version:**  2.1.1

**Code**

```ts
const evil = {
    hasOwnProperty: () => true
};

evil.hasOwnProperty("foo"); // true
hasOwnProperty.call(evil, "foo"); // false
```

**Expected behavior:**

Nothing should be flagged as improper, since this is all valid code

**Actual behavior:**

`hasOwnProperty` is flagged as not existing. It technically does exist as part of the global objects in both Node and standard browsers.

Shouldn't it be defined?

**Comment by aluanhaddad** [comment]
It seems to me that if you want to write something that awful, you can use declare global.
```ts
declare global {
  function hasOwnProperty(key: string | number | symbol): boolean;
}
```

**Comment by aluanhaddad** [comment]
Actually, that actually allows you to make it less awful
```ts
declare global {
  const hasOwnProperty: { 
    call: (key: string | number | symbol): boolean
  };
}
```

**Comment by mhegazy** [comment]
I do not see any reference in the spec to a global `hasOwnProperty`. this sees to be the `window.hasOwnProperty` in a browser environment. The TypeScript standard library deliberately does not represent `Window` as a global-polluter to restrict global variable use.

**Comment by JoshuaKGoldberg** [comment]
I generally agree with and like your rule of not polluting globals with `Window` shenanigans... but this seems like a good practice case. How else do you check for the *existence* of keys in an object? 

As a side note, since SEO seems to have bumped this issue thread near the top of some searches, here's a handy list for the alternatives & why I'm complaining about them:
* `window.hasOwnProperty.call(container, key)` doesn't work in Node, and is a little verbose
* `container.hasOwnProperty("key")` can be mocked out *(see above)*
* `"key" in container` returns `true` for properties on the prototype chain
* `container[key]` is truthiness instead of existence.

**Comment by kpreisser** [comment]
What about `Object.prototype.hasOwnProperty.call(container, key)`? Though a little verbose, I think it should work in all cases.

**Comment by JoshuaKGoldberg** [comment]
`{}.hasOwnProperty.call(container, key)` works top I suppose. Still seems unnecessarily verbose.
